docs/api-reference/tasks-api.md

```diff
@@ -25,77 +25,1454 @@ sidebar_label: Tasks
+            "requiredCapacity": 1
+        }
+    }
+  }
+  ```
+
+</details>
+
+### Get task status
+
+#### URL
+<code class="getAPI">GET</code> `/druid/indexer/v1/task/{taskId}/status`
+
+Retrieves the status of a task given the task ID. It returns a JSON object with the task's current state, task type, datasource, and other relevant metadata.
+
+#### Responses 
+<!--DOCUSAURUS_CODE_TABS-->
 
-`GET /druid/indexer/v1/pendingTasks`
+<!--200 SUCCESS-->
+<br/>
+*Successfully retrieved task status* 
+<!--404 NOT FOUND-->
+<br/>
+*Cannot find task with ID* 
+<!--END_DOCUSAURUS_CODE_TABS-->
 
-Retrieve list of pending tasks. Equivalent to `/druid/indexer/v1/tasks?state=pending`.
+---
+
+#### Sample request
+
+The following examples shows how to retrieve the status of a task with the specified ID `query-223549f8-b993-4483-b028-1b0d54713cad`.
+
+<!--DOCUSAURUS_CODE_TABS-->
+
+<!--cURL-->
+```shell
+curl "{domain}/druid/indexer/v1/task/query-223549f8-b993-4483-b028-1b0d54713cad/status"
+```
+<!--HTTP-->
+```HTTP
+GET /druid/indexer/v1/task/query-223549f8-b993-4483-b028-1b0d54713cad/status HTTP/1.1
+Host: {domain}
+```
+<!--END_DOCUSAURUS_CODE_TABS-->
+
+
+#### Sample response
+<details>
+  <summary>Click to show sample response</summary>
 
-`GET /druid/indexer/v1/task/{taskId}`
+  ```json
+  {
+    'task': 'query-223549f8-b993-4483-b028-1b0d54713cad',
+    'status': {
+      'id': 'query-223549f8-b993-4483-b028-1b0d54713cad',
+      'groupId': 'query-223549f8-b993-4483-b028-1b0d54713cad',
+      'type': 'query_controller',
+      'createdTime': '2023-06-22T22:11:28.367Z',
+      'queueInsertionTime': '1970-01-01T00:00:00.000Z',
+      'statusCode': 'RUNNING',
+      'status': 'RUNNING',
+      'runnerStatusCode': 'RUNNING',
+      'duration': -1,
+      'location': {'host': 'localhost', 'port': 8100, 'tlsPort': -1},
+      'dataSource': 'wikipedia_api',
+      'errorMsg': None
+    }
+  }
+  ```
 
-Retrieve the 'payload' of a task.
+</details>
 
-`GET /druid/indexer/v1/task/{taskId}/status`
+### Get task segments
 
-Retrieve the status of a task.
+#### URL
 
-`GET /druid/indexer/v1/task/{taskId}/segments`
+<code class="getAPI">GET</code> `/druid/indexer/v1/task/{taskId}/segments`
 
 > This API is deprecated and will be removed in future releases.
 
-Retrieve information about the segments of a task.
+Retrieves information about segments generated by the task given the task ID. To hit this endpoint, make sure to enable the audit log config on the Overlord with `druid.indexer.auditLog.enabled = true`.
+
+In addition to enabling audit logs, configure a cleanup strategy to prevent overloading the metadata store with old audit logs and cause performance issues. You can enable automated cleanup of audit logs on the Coordinator with `druid.coordinator.kill.audit.on` or export the audit logs to external storage. For reference, see [Audit records](../operations/clean-metadata-store.md#audit-records).
```

Comment:
```suggestion
In addition to enabling audit logs, configure a cleanup strategy to prevent overloading the metadata store with old audit logs which may cause performance issues. To enable automated cleanup of audit logs on the Coordinator, set `druid.coordinator.kill.audit.on`. You may also manually export the audit logs to external storage. For more information, see [Audit records](../operations/clean-metadata-store.md#audit-records).
```
I assume export is a manual process and not automated cleanup?